### ISSUE TITLE
Agent not reporting logs due to dd-agent missing systemd-journal group

### ISSUE THREAD

**Issue by endorama** [DataDog/ansible-datadog#146]
Hello, we are using this role for deploying `datadog` agents on our instances.

While configuring `journald` log collection on 2 different EC2 instances, I found a strange behaviour: *instance-2* was correctly reporting logs, *instance-1* was not. The same playbook has been applied to both instances, using this role.

The datadog configuration applied is:
```yaml
datadog_api_key: <omitted>
datadog_config:
  logs_enabled: true
datadog_checks:
  journald:
    logs:
    - type: journald
      include_units:
        - our-custom.service
```

Both agent are starting without errors. *instance-1* agent logs report tailing from journald as expected:
```
instance-1 systemd[1]: Started "Datadog Agent".
.
.
.
... INFO | (logs.go:29 in Start) | Starting logs-agent
... INFO | (tailer.go:45 in Start) | Start tailing journal  default
.
.
.
```

After some *hours* (:persevere:) of debugging, reading and re reading the docs I found a subtle difference between the two instances:
- *instance-2*
    ```
    # id dd-agent
    uid=114(dd-agent) gid=117(dd-agent) groups=117(dd-agent),101(systemd-journal)
    ```
- *instance-1*
    ```
    # id dd-agent
    uid=114(dd-agent) gid=117(dd-agent) groups=117(dd-agent)
    ```

My bad as I probably add it manually only on one of the two, and as your [installation documentation](https://docs.datadoghq.com/integrations/journald/#installation) for journald log collection states:
> Add the dd-agent user to the systemd-journal group by running:

What I actually didn't expected was:
- this step to be out of this ansible role
- no documentation of this **mandatory** step in the readme ( or on the online docs ); I also suspect that other integration would require other groups to be assigned to the `dd-agent` user
- the logs showing "Start tailing journal  default", without any kind of error about the impossibility to tail journald

I don't know what would be the best way to inform users of this role about this, but would be nice if you could do something about as it has been a really difficult and long debugging session.

Thank you :slightly_smiling_face:

**Comment by mgu** [comment]
what could be great is a way to add dd-agent user to extra groups (`systemd-journal`,  `adm`, ...) in this role

**Comment by NilsBunge** [comment]
Hello @endorama and @mgu ,

Thanks for reaching out and for your suggestions. 

By default, the agent tries to connect to the journal and reads what he can. In your case it was nothing but as it could read the file, it did not raise an error in the status.
We will see if we can improve this behaviour to avoid long debugging sessions in the future.

We are also looking at the different options to add the `dd-agent` user in the `systemd-journal` group directly in the ansible role as you suggested (and add some documentation for this).

**Comment by rromanchuk** [comment]
For now i'm going to do this

```yml
- name: Setup base datadog agent
  hosts: ec2
  roles:
    - role: Datadog.datadog
      become: yes
      datadog_api_key: "redacted" # On Ansible < 1.9, use "sudo: yes" instead of "become: yes"
      datadog_disable_untracked_checks: false
  tasks:
    - name: Add the dd-agent user to the systemd-journal
      become: yes
      user:
        name: dd-agent
        group: systemd-journal
        append: yes

**Comment by rromanchuk** [comment]
Uggghhhh, just got burned by this again, FYI  it's `groups: systemd-journal` not `group: systemd-journal`

https://docs.ansible.com/ansible/devel/modules/user_module.html

```yaml
- name: Add the dd-agent user to the systemd-journal
      become: yes
      user:
        name: dd-agent
        group: systemd-journal
        append: yes
```

**Comment by endorama** [comment]
Update after #194 has been merged.

Official solution is to use the `datadog_additional_groups` variable to add `dd-agent` to `systemd-journal`.  
[Pretty much the same](https://github.com/DataDog/ansible-datadog/pull/194/files#diff-006bb07c7c283d1d4197a91c61d243f4) as @rromanchuk solution without the hassle of a separate task.

To be able to use this feature you should update to version `3.0.0` of this role.

I'll close this, as an official solution has been released an no further work is planned.
